File: DictionaryProject/Commands/MarkCommand.vb
<Highlight(RegEx:="^\s*mark ", Color:="Green")> _
Public Class MarkCommand
    Inherits CommandBase

    Private _Name As String
    Public Property Name() As String
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
        End Set
    End Property

    Private _Value As String
    Public Property Value() As String
        Get
            Return _Value
        End Get
        Set(ByVal value As String)
            _Value = value
        End Set
    End Property


    Public Overrides Sub Execute(ByVal context As Context)
        context.Mark(Name, Value)
    End Sub

    Public Overrides Sub LoadCommand(ByVal project As Project, ByVal reader As System.IO.TextReader, ByVal line As String, ByRef lineNumber As Integer)
        MyBase.LoadCommand(project, reader, line, lineNumber)

        Dim parts As List(Of String) = ProjectSerializer.ReadTokens(line)

        If parts.Count <> 3 Then
            project.Warnings.Add(String.Format("Line {0}: The mark command requires 2 arguments.", lineNumber))
        Else
            _Name = parts(1)
            _Value = parts(2)
        End If
    End Sub

    Public Overrides Sub WriteCommand(ByVal writer As System.IO.TextWriter)
        writer.WriteLine("Mark {0} {1}", _Name, _Value)
    End Sub

    Public Overrides Sub CheckSanity(ByVal project As Project)

    End Sub
End Class

File: monowordbuilder/wordbuilderbase/Commands/MarkCommand.cs
using System.Collections.Generic;

namespace Whee.WordBuilder.Model.Commands
{
	[Highlight(RegEx = "^\\s*mark ", Color = "Green")]
	public class MarkCommand : CommandBase
	{
	
		private string _Name;
		public string Name {
			get { return _Name; }
			set { _Name = value; }
		}
	
		private string _Value;
		public string Value {
			get { return _Value; }
			set { _Value = value; }
		}
	
	
		public override void Execute(Context context)
		{
			context.Mark(Name, Value);
		}

        public override void LoadCommand(Whee.WordBuilder.ProjectV2.IProjectSerializer serializer)
        {
            m_lineNumber = serializer.LineNumber;

            ProjectV2.Token name = serializer.ReadTextToken(this);
            if (name != null)
            {
                Name = name.Text;
                name.Type = Whee.WordBuilder.ProjectV2.TokenType.Name;

                ProjectV2.Token value = serializer.ReadTextToken(this);
                if (value != null)
                {
                    Value = value.Text;

                    if (serializer.ReadTextToken(this) != null)
                    {
                        serializer.Warn("The mark command requires 2 arguments.", this);
                    }
                }
                else
                {
                    serializer.Warn("The mark command requires 2 arguments.", this);
                }
            }
            else
            {
                serializer.Warn("The mark command requires 2 arguments.", this);
            }
        }

		public override void WriteCommand(System.IO.TextWriter writer)
		{
			writer.WriteLine("Mark {0} {1}", _Name, _Value);
		}

        public override void CheckSanity(Project project, Whee.WordBuilder.ProjectV2.IProjectSerializer serializer)
        {
        }
    }
}